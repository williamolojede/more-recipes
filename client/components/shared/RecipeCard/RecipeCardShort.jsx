import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const RecipeCardShort = ({
  id,
  name,
  removeRecipe,
  type
}) => (
  <li className="col s12 m6 l4 x4 recipes-list-item">
    <div className="card">
      <Link to={`/recipe/${id}`} className="recipes-list-item__name">
        {name}
      </Link>
      <div className="recipes-list-item__actions">
        {
          type !== 'favorite' &&
          <Link
            to={`/user/recipes/${id}/edit`}
            className="recipes-list-item__actions-edit"
          >
            <i className="mdi mdi-pencil" />
          </Link>
        }
        <button
          onClick={() => removeRecipe(id)}
          className="recipes-list-item__actions-delete"
        >
          <i className="mdi mdi-delete" />
        </button>
      </div>
    </div>
  </li>
);

RecipeCardShort.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  removeRecipe: PropTypes.func.isRequired,
  type: PropTypes.oneOf(['favorite', 'personal']).isRequired
};

export default RecipeCardShort;
