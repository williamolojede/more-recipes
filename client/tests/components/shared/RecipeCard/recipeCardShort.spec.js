import React from 'react';
import RecipeCardShort from '../../../../components/shared/RecipeCard/RecipeCardShort.jsx';
import recipesMock from '../../../__mocks__/recipes.mock';

function setup(type) {
  const {
    id,
    name
    // upVoteCount,
    // downVoteCount,
    // favoriteCount,
    // viewCount
  } = recipesMock.recipes[5];

  const props = {
    id,
    name,
    removeRecipe: jest.fn(),
    type
  };
  const wrapper = shallow(<RecipeCardShort {...props} />);

  return {
    wrapper,
    props
  };
}

describe('Recipe Card Info', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup('personal');
    expect(wrapper).toMatchSnapshot();
  });

  it('should call a function to delete a recipe when its clicked', () => {
    const { wrapper, props } = setup('personal');
    const deleteButton = wrapper.find('button');
    expect(deleteButton.length).toBe(1);
    const removeRecipeSpy = jest.spyOn(wrapper.instance().props, 'removeRecipe');

    deleteButton.simulate('click');
    expect(removeRecipeSpy).toHaveBeenCalledTimes(1);
    expect(removeRecipeSpy).toHaveBeenCalledWith(props.id);
  });
});
