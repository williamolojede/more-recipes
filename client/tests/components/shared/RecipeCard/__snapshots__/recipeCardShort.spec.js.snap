// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card Info should render with right amount of elements 1`] = `
<li
  className="col s12 m6 l4 x4 recipes-list-item"
>
  <div
    className="card"
  >
    <Link
      className="recipes-list-item__name"
      replace={false}
      to="/recipe/28"
    >
      Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref
    </Link>
    <div
      className="recipes-list-item__actions"
    >
      <Link
        className="recipes-list-item__actions-edit"
        replace={false}
        to="/user/recipes/28/edit"
      >
        <i
          className="mdi mdi-pencil"
        />
      </Link>
      <button
        className="recipes-list-item__actions-delete"
        onClick={[Function]}
      >
        <i
          className="mdi mdi-delete"
        />
      </button>
    </div>
  </div>
</li>
`;
